subprojects/dependency-management/src/main/java/org/gradle/internal/component/local/model/DefaultLocalConfigurationMetadata.java

```diff
@@ -162,67 +94,34 @@ private DefaultLocalConfigurationMetadata(
         this.artifacts = artifacts;
     }
 
-    /**
-     * Creates a calculated value container which lazily constructs this configuration's artifacts
-     * by traversing all configurations in the hierarchy and collecting their artifacts.
-     */
-    private static CalculatedValue<ImmutableList<LocalComponentArtifactMetadata>> getLazyArtifacts(
-        List<PublishArtifact> sourceArtifacts, String name, String description, Set<String> hierarchy,
-        ModelContainer<?> model, CalculatedValueContainerFactory factory, LocalComponentMetadata component
-    ) {
-        return factory.create(Describables.of(description, "artifacts"), context -> {
-            if (sourceArtifacts.isEmpty() && hierarchy.isEmpty()) {
-                return ImmutableList.of();
-            } else {
-                return model.fromMutableState(m -> {
-                    Set<LocalComponentArtifactMetadata> result = new LinkedHashSet<>(sourceArtifacts.size());
-                    for (PublishArtifact sourceArtifact : sourceArtifacts) {
-                        // The following line may realize tasks, so we wrap this code in a CalculatedValue.
-                        result.add(new PublishArtifactLocalArtifactMetadata(component.getId(), sourceArtifact));
-                    }
-                    for (String config : hierarchy) {
-                        if (config.equals(name)) {
-                            continue;
-                        }
-                        // TODO: Deprecate the behavior of inheriting artifacts from parent configurations.
-                        LocalConfigurationMetadata parent = component.getConfiguration(config);
-                        result.addAll(parent.prepareToResolveArtifacts().getArtifacts());
-                    }
-                    return ImmutableList.copyOf(result);
-                });
-            }
-        });
-    }
-
     @Override
     public LocalConfigurationMetadata copy(Transformer<LocalComponentArtifactMetadata, LocalComponentArtifactMetadata> artifactTransformer) {
```

Comment:
❓  This is a `copy` method that takes a `Transformer` argument to apply to the resolved artifacts (which might not be resolved yet) and doesn't even guarantee in its signature to return the same type as it is defined in.  I don't think the result can be thought of as a copy.  

Can this method be named differently, and given some javadoc explaining its purpose?

Reply:
Added javadoc and renamed to `copyWithTransformedArtifacts`